version: "3.7"

services:
  nginx:
    image: nginx:1.19.5

    ports:
      - 80:80

    networks:
      - gis_4_2_network

    volumes:
      - /home/ljk8190/nginx.conf:/etc/nginx/nginx.conf
      - gis_4_2_static:/data/static
      - gis_4_2_media:/data/media

  django:
    image: gis_4_2_django:1.0.4

    networks:
      - gis_4_2_network

    volumes:
      - gis_4_2_static:/home/gis4_2/staticfiles/
      - gis_4_2_media:/home/gis4_2/media/

    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD

  mariadb:
    image: mariadb:10.5

    networks:
      - gis_4_2_network

    volumes:
      - gis_4_2_mariadb:/var/lib/mysql

    secrets:
      - MARIADB_ROOT_PASSWORD
      - MARIADB_USER
      - MARIADB_PASSWORD

    environment:
      MARIADB_ROOT_PASSWORD_FILE: /run/secrets/MARIADB_ROOT_PASSWORD
      MARIADB_DATABASE: django
      MARIADB_USER_FILE: /run/secrets/MARIADB_USER
      MARIADB_PASSWORD_FILE: /run/secrets/MARIADB_PASSWORD


networks:
  gis_4_2_network:

volumes:
  gis_4_2_static:
  gis_4_2_media:
  gis_4_2_mariadb:

secrets:
  DJANGO_SECRET_KEY:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true